### ISSUE TITLE
False positive: if docblock has @return static, then it should not display "XXX can be declared as type hint"

### ISSUE THREAD

**Issue by thomasphansen** [kalessil/phpinspectionsea#1533]
| Subject        | Details                                                                       |
| :------------- | :---------------------------------------------------------------------------- |
| Plugin         | Php Inspections (EA Extended), 4.0.3 Jan 06 2020 |
| Language level | PHP 7.3                        |

![Screenshot from 2020-03-19 14-52-02](https://user-images.githubusercontent.com/5396791/77074563-50b07d80-69f1-11ea-839c-e8359f04af10.png)

Code sample
--

```
class MyTest
{
    /**
     * @return static
     */
    public function test()
    {
        return $this;
    }
}
```

Current behavior
--

[EA] Complains for the method method test() : "[EA] ': MyTest' can be declared as type hint."

Expected behavior
--

There should be no complain. Declaring MyTest as type hint, whether not preventing descendant classes to return $this, would cause trouble on their usage. Look at the following code:
```
class MyTest2 extends MyTest
{
    public function test2(): void
    {
        echo 'bla';
    }
}

$oTest = new MyTest();
$oTest
    ->test()
    ->test2(); // Would be undefined: "Method 'test2' not found in MyTest"
```
Php 8 will have the "static" return type, which will solve this question for good. But right now, EA is suggesting an invalid fix, as demonstrated above.

Environment details
---

PhpStorm 2020.1 EAP
Build #PS-201.6251.26, built on March 12, 2020
PhpStorm EAP User
Expiration date: April 11, 2020
Runtime version: 11.0.6+8-b765.15 amd64
VM: OpenJDK 64-Bit Server VM by JetBrains s.r.o
Linux 5.3.0-40-generic
GC: ParNew, ConcurrentMarkSweep
Memory: 4029M
Cores: 8
Registry: run.processes.with.pty=TRUE, editor.old.tab.painting=true
Non-Bundled Plugins: com.jetbrains.upsource, color.scheme.IdeaLight, com.intellij.properties, navigate-from-literal, org.intellij.RegexpTester, com.dmarcotte.handlebars, com.dryabov.phpStorm.phpregexp, com.kalessil.phpStorm.phpInspectionsEA, org.zalando.intellij.swagger
Current Desktop: X-Cinnamon
